app/lib/actions/katello/orphan_cleanup/remove_orphans.rb

```diff
@@ -7,27 +7,16 @@ class RemoveOrphans < Actions::Base
         end
         def plan(proxy)
           sequence do
+            if proxy.pulp_primary?
+              ::Katello::RootRepository.orphaned.destroy_all
```

Comment:
Any reason why this was pulled out of the run phase?

Reply:
Not really..This was the only operation in the run phase after refactoring the content unit deletion..and this doesn't seem like something that needs to be in the run phase..

Reply:
If there's a DB operation, should try not to put it in the run phase because `run` doesn't run in a transaction :)

Reply:
Would a heavy DB operation in the planning phase eat up a Puma worker? Also we shouldn't need transactions here at least since we're deleting orphans.  But even if we did need one, couldn't we do a `transaction` block?

Reply:
> I'm a little confused because we stick heavy DB ops in tasks in other places

Usually they're in `plan` or `finalize` though, right? Those are fine because they run in transactions by default.

Reply:
Ack to that..We tend to put only things that need API calls to other products in run phase generally. For this task, I'd be ok keeping this in run phase over the plan phase. Any failures on this are set to rescue=skip and also we do want to still plan other pulp orphan deletion actions even if the orphan deletion in katello fails. A plan phase failure wouldn't plan those pulp tasks. 

We could put this in finalize if we want. Let me know if we feel strongly about that and I can update this.

Reply:
Makes sense to me. I trust your judgment to choose the phase correctly :)